stactools_landsat/stactools/landsat/utils.py

```diff
@@ -102,40 +103,33 @@ def transform_stac_to_stac(item: Item,
 
     if enable_proj:
         try:
-            # If we can load the blue band, use it to add proj information
-            if item.assets.get("SR_B2.TIF"):
-                asset = item.assets["SR_B2.TIF"]
-            # SR_B10 is a fallback for SR_B2
-            elif item.assets.get("SR_B10.TIF"):
-                asset = item.assets["SR_B10.TIF"]
-            else:
-                raise ValueError('Asset SR_B2.TIF or SR_B2.TIF required')
-
-            with rasterio.open(asset.href) as opened_asset:
-                shape = [opened_asset.height, opened_asset.width]
-                transform = opened_asset.transform
-                crs = opened_asset.crs.to_epsg()
-
-            # Now we have the info, we can make the fields
-            item.ext.enable("projection")
-            item.ext.projection.epsg = crs
-
+            retrieved_tile_info = None
             for name, asset in item.assets.items():
-                if asset.media_type == "image/vnd.stac.geotiff; cloud-optimized=true":
+                if "geotiff" in asset.media_type:
+                    if not retrieved_tile_info:
+                        with rasterio.open(asset.href) as opened_asset:
+                            shape = opened_asset.shape
+                            transform = opened_asset.transform
+                            crs = opened_asset.crs.to_epsg()
+                            retrieved_tile_info = True
+
                     item.ext.projection.set_transform(transform, asset=asset)
                     item.ext.projection.set_shape(shape, asset=asset)
```

Comment:
It looks like the projection extension is being set multiple times here (one for each asset that has "geotiff" in the media type). This should probably be moved below this loop and only set once (i.e. to https://github.com/stac-utils/stactools/pull/100/files/8d3c80a8bfb290317f96637a58e6ed2d7f4fa412#diff-408a908da00c6d195d2d9215a17f6d21d27b3e39e63c4a017b4e562727356363R122).

Reply:
Hi Pete,

In this case 
```
shape = opened_asset.shape
transform = opened_asset.transform
crs = opened_asset.crs.to_epsg()
```
Will be set just the first time that the code finds a geotiff, it happens because of the flag ```retrieved_tile_info = True``` which won't allow to set that again. I added this code inside of the loop, because otherwise I'd have to loop twice.